drivers/note/notesched_driver.c

```diff
@@ -0,0 +1,393 @@
+                                 FAR struct tcb_s *tcb)
+{
+  notesched_common(tcb, NOTE_RESUME, 0);
+}
+
+static void notesched_cpu_resume(FAR struct note_driver_s *drv,
+                                 FAR struct tcb_s *tcb, int cpu)
+{
+  notesched_common(tcb, NOTE_RESUME, cpu);
+}
+
+static void notesched_cpu_resumed(FAR struct note_driver_s *drv,
+                                  FAR struct tcb_s *tcb)
+{
+  notesched_common(tcb, NOTE_RESUME, 0);
+}
+#endif
+#endif
+
+#ifdef CONFIG_SCHED_INSTRUMENTATION_PREEMPTION
+static void notesched_premption(FAR struct note_driver_s *drv,
+                                FAR struct tcb_s *tcb, bool locked)
+{
+  notesched_common(tcb, locked ? NOTE_PREEMPT_LOCK : NOTE_PREEMPT_UNLOCK, 0);
+}
+#endif
+
+#ifdef CONFIG_SCHED_INSTRUMENTATION_CSECTION
+static void notesched_csection(FAR struct note_driver_s *drv,
+                               FAR struct tcb_s *tcb, bool enter)
+{
+  notesched_common(tcb,
+                   enter ? NOTE_CSECTION_ENTER : NOTE_CSECTION_LEAVE, 0);
+}
+#endif
+
+#ifdef CONFIG_SCHED_INSTRUMENTATION_SPINLOCKS
+static void notesched_spinlock(FAR struct note_driver_s *drv,
+                               FAR struct tcb_s *tcb,
+                               FAR volatile void *spinlock, int type)
+{
+  notesched_common(tcb, type, (uintptr_t)spinlock);
+}
+#endif
+
+#ifdef CONFIG_SCHED_INSTRUMENTATION_IRQHANDLER
+static void notesched_irqhandler(FAR struct note_driver_s *drv, int irq,
+                                 FAR void *handler, bool enter)
+{
+  notesched_common(this_task(), enter ? NOTE_IRQ_ENTER : NOTE_IRQ_LEAVE,
+                   (uintptr_t)irq);
+}
+#endif
+
+#ifdef CONFIG_SCHED_INSTRUMENTATION_SYSCALL
+static void notesched_syscall_enter(FAR struct note_driver_s *drv, int nr,
+                                    int argc, va_list *ap)
+{
+  notesched_common(this_task(), NOTE_SYSCALL_ENTER, (uintptr_t)nr);
+}
+
+static void notesched_syscall_leave(FAR struct note_driver_s *drv, int nr,
+                                    uintptr_t result)
+{
+  notesched_common(this_task(), NOTE_SYSCALL_LEAVE, (uintptr_t)nr);
+}
+#endif
+
+/****************************************************************************
+ * Public Functions
+ ****************************************************************************/
+
+/****************************************************************************
+ * Name: notesched_register
+ ****************************************************************************/
+
+int notesched_register(void)
+{
+  return note_driver_register(&g_notesched.driver);
+}
+
+/****************************************************************************
+ * Name: notesched_dump
+ ****************************************************************************/
+
+void notesched_dump(void)
+{
+  size_t i;
+  struct note_filter_mode_s old;
+  size_t index = g_notesched.index;
+  FAR struct notesched_chunk_s *note;
+
+  sched_note_filter_mode(&old, NULL);
```

Comment:
why need change? the will impact other note driver.

Reply:
This is to avoid logging syslog in notesched_dump, I have changed it to lib_sprint